docs/architecture/adr-022-per-subspace-token-factory.md

```diff
@@ -0,0 +1,228 @@
+# ADR 022: Per-subspace token factory
+
+## Changelog
+- April 28th, 2023: First draft;
+
+## Status
+
+Draft
+
+## Abstract
+
+This ADR introduces a new feature that enables the subspace admins to create, mint and burn a subspace related
+coin. 
+
+## Context
+
+Desmos is a social network protocol that allows users to create, share, and engage with content 
+on a decentralized platform. 
+Currently, to post or interact with the content in a subspace you need to pay the gas fee using DSM. 
+This have the consequence of not giving any financial incentive on creating a social network on Desmos.
+
+## Decision
+
+We will wrap the Osmosis [Token Factory module](https://docs.osmosis.zone/osmosis-core/modules/tokenfactory/)
+with the following modifications:
+1. Instead of use the token creator address to compose the coin denom we will use the subspace treasury address
+   as following: `factory/{trasury_address}/subdenom`;
+2. All the operations of `CreateDenom`, `Mint`, `Burn`, `SetDenomMetadata`, `SetBeforeSendHook` and
+   `ForceTransfer` can only be performed by subspace treasury;
+3. The `CreateDenom` action will burn the coins instead of send them to the community pool.  
+
+With this module subspace admins will be able to create a coin that can be used to pay for 
```

Comment:
The intension is having custom tokenomic for subspace easily, rather than paying subspace transactions.